hoot-core/src/main/cpp/hoot/core/ops/CalculateStatsOp.cpp

```diff
@@ -469,6 +475,62 @@ void CalculateStatsOp::apply(const shared_ptr<OsmMap>& map)
     _stats.append(
       SingleStat("Percentage of Unmatched Buildings", percentageOfTotalBuildingsUnconflated));
 
```

Comment:
I haven't looked into the caveats, but the list of feature specific stats is growing pretty quickly -- mostly w/ cut/paste. Would it be reasonable to parameterize a function that works for each type. E.g.

```
_generateFeatureStats(_stats, "Building", new BuildingCriterion());
_generateFeatureStats(_stats, "Highway", new HighwayCriterion());
_generateLinearFeatureStats(_stats, "Highway", new WaterwayCriterion());
_generateFeatureStats(_stats, "Waterway", new WaterwayCriterion());
_generateLinearFeatureStats(_stats, "Waterway", new WaterwayCriterion());
```

This would make the code a bit easier to maintain.